third_party/tonic/dart_args.h

```diff
@@ -238,6 +250,292 @@ void DartCallConstructor(Sig func, Dart_NativeArguments args) {
   wrappable->AssociateWithDartWrapper(wrapper);
 }
 
+// Template voodoo to automatically setup static entry points for FFI Native
+// functions.
+// Entry points for instance methods take the instance as the first argument and
+// call the given method with the remaining arguments.
+// Arguments will automatically get converted to and from their FFI
+// representations with the DartConverter templates.
+template <typename Object, typename Signature, Signature Method>
+struct FfiDispatcher;
+
+// Recursively add the FFI representation of arguments to the stream,
+// serialising them into a comma separated list.
+template <typename Arg, typename... Args>
+void WriteFfiArguments(std::ostringstream* stream) {
+  *stream << tonic::DartConverter<typename std::remove_const<
+      typename std::remove_reference<Arg>::type>::type>::GetFfiRepresentation();
+  if constexpr (sizeof...(Args) > 0) {
+    *stream << ", ";
+    WriteFfiArguments<Args...>(stream);
+  }
+}
+
+// Recursively add the Dart representation of arguments to the stream,
+// serialising them into a comma separated list.
+template <typename Arg, typename... Args>
+void WriteDartArguments(std::ostringstream* stream) {
+  *stream << tonic::DartConverter<
+      typename std::remove_const<typename std::remove_reference<Arg>::type>::
+          type>::GetDartRepresentation();
+  if constexpr (sizeof...(Args) > 0) {
+    *stream << ", ";
+    WriteDartArguments<Args...>(stream);
+  }
+}
+
+// Recursively and together whether each argument is allowed in a leaf call.
+template <typename Arg, typename... Args>
+bool AllowedInLeafCall() {
+  bool result = tonic::DartConverter<typename std::remove_const<
+      typename std::remove_reference<Arg>::type>::type>::AllowedInLeafCall();
+  if constexpr (sizeof...(Args) > 0) {
+    result &= AllowedInLeafCall<Args...>();
+  }
+  return result;
+}
+
+// Match `Return Function(...)`.
+template <typename Return, typename... Args, Return (*Function)(Args...)>
```

Comment:
Should `*Function` be `*function` because it is not a type but a function address?